src/main/java/com/google/jenkins/plugins/computeengine/PreemptedCheckCallable.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.jenkins.plugins.computeengine;
+
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpHeaders;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import hudson.model.TaskListener;
+import java.io.IOException;
+import jenkins.security.MasterToSlaveCallable;
+import org.apache.commons.io.Charsets;
+import org.apache.commons.io.IOUtils;
+
+/**
+ * Slave callback class checking if instance was preempted. All of code here is serialized and
+ * executed on node.
+ */
+final class PreemptedCheckCallable extends MasterToSlaveCallable<Boolean, IOException> {
+  private static final String METADATA_SERVER_URL =
+      "http://metadata.google.internal/computeMetadata/v1/instance/preempted?wait_for_change=true";
+
+  private final TaskListener listener;
+
+  /**
+   * Callback constructor.
+   *
+   * @param listener Node listener on which we can add extra information from slave side.
+   */
+  PreemptedCheckCallable(TaskListener listener) {
+    this.listener = listener;
+  }
+
+  /**
+   * Actual callback code, executed on node side. Checks in Google metadata server if instance was
+   * preempted.
+   *
+   * <p>See
+   * https://cloud.google.com/compute/docs/instances/create-start-preemptible-instance#detecting_if_an_instance_was_preempted
+   *
+   * @return True if node was preempted.
+   * @throws IOException Exception when calling Google metadata API
+   */
+  @Override
+  public Boolean call() throws IOException {
+    HttpTransport transport = new NetHttpTransport();
```

Comment:
Throughout the plugin we maintain abstraction layers for calls to the GCP API through the client classes. This looks like a good candidate for a MetadataClient abstraction.

Reply:
Because there is almost 0 logic here.. and noone else will ever use it..
And its serialised and send to slaves I want to keep it as simple as possible..

I can split it into 2 methods for readability..